### PR TITLE
New Search Results

### PR DESCRIPTION
**Pull request by amielmon**
Still missing:
- Campuses
- Accelerated, Honors, etc.

Mobile and desktop search results
**Desktop Result (shitty screenshot)**
Screen Shot 2020-06-25 at 2 18 32 PM
**Mobile Result (subscribe button WIP)**
Screen Shot 2020-06-25 at 2 21 27 PM
 **Notif States** 
Screen Shot 2020-06-25 at 2 19 20 PM
Screen Shot 2020-06-25 at 2 19 31 PM
Screen Shot 2020-06-25 at 2 19 55 PM
Screen Shot 2020-06-25 at 2 19 41 PM
Screen Shot 2020-06-25 at 2 20 10 PM
**Expanded mobile description** 
Screen Shot 2020-06-25 at 2 22 06 PM

### CODE REVIEW

**Comment by ryandrew14** [comment]
Is there CD on Search? If not could u put screenshots maybe

**Comment by amielmon** [comment]
> Is there CD on Search? If not could u put screenshots maybe

Okey boss

**Comment by ryandrew14** [comment]
https://open.spotify.com/track/3avYqdwHKEq8beXbeWCKqJ?si=Kb01SELSSdaPTeXZcE1IWQ

**Comment by amielmon** [comment]
> https://open.spotify.com/track/3avYqdwHKEq8beXbeWCKqJ?si=Kb01SELSSdaPTeXZcE1IWQ

OOp finally put screenshots. Lmk if you want to see anything specific.

**Comment by ryandrew14** [comment]
there are really 1400 actual loc in this T_T ill read it later. screenshots look great tho

**Comment by beccajohnson17** [comment]
I think this looks great! :)

I have one question about the screenshots/minor feedback:

1) do you think we need to repeat the times? I was thinking that if there was more than one time for a section we could stack the days of the week bar and highlight only the days that match that time, but otherwise, I think having one time displayed would be sufficient.

2) I think aligning the results with their headers (CRN, Professors, show more results, etc.) could be pretty clean.

3) for the days of the week bar, it looks like the lines are a little thick on the inside (for example, between S and M and M and T). I think having them overlap on the inside would be a bit more cohesive.

**Review by beccajohnson17 — COMMENTED** [review]
I think this looks great! :)
I have one question about the screenshots/minor feedback:

1) do you think we need to repeat the times? I was thinking that if there was more than one time for a section we could stack the days of the week bar and highlight only the days that match that time, but otherwise, I think having one time displayed would be sufficient.

2) I think aligning the results with their headers (CRN, Professors, show more results, etc.) could be pretty clean.

3) for the days of the week bar, it looks like the lines are a little thick on the inside (for example, between S and M and M and T). I think having them overlap on the inside would be a bit more cohesive.

4) We have a spec for logging into facebook, but if you think the current one we have is preferred we can keep it as is

**Comment by amielmon** [comment]
> I think this looks great! :)
> I have one question about the screenshots/minor feedback:
> 
> 1. do you think we need to repeat the times? I was thinking that if there was more than one time for a section we could stack the days of the week bar and highlight only the days that match that time, but otherwise, I think having one time displayed would be sufficient.
> 2. I think aligning the results with their headers (CRN, Professors, show more results, etc.) could be pretty clean.
> 3. for the days of the week bar, it looks like the lines are a little thick on the inside (for example, between S and M and M and T). I think having them overlap on the inside would be a bit more cohesive.
> 4. We have a spec for logging into facebook, but if you think the current one we have is preferred we can keep it as is

I tried implementing the facebook button spec several ways but would always run into a different problem. Mind you, the button comes directly from Facebook. Rendering the new button over the opaque existing button didn't work well do to resizing and positioning issues ('is this you' text that appears if you've logged in before would make the touch area variable and resizing both the button and container was a pain). Trying to hide the button and programmatically click it with a ref didn't work because the actual clickable element was deeply nested in the component returned by Facebook. Because of this, I opted to keep the original.

I will make changes to the table and weekday boxes.

File: frontend/components/ResultsPage/Results/SearchResult.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+import useResultRequisite from './useResultRequisite'
+import useUserChange from './useUserChange';
+import useShowAll from './useShowAll';
+
+interface SearchResultProps {
+  aClass: Course,
+}
+export default function SearchResult({ aClass } : SearchResultProps) {
+  const optionalDisplay = useResultRequisite()
+  const userIsWatchingClass = useUserChange(aClass)
```
Comment (dajinchu) [inline frontend/components/ResultsPage/Results/SearchResult.tsx:17]: nice. I think we could probably later get rid of the `user` module entirely, but good enough for now

**Review by dajinchu — APPROVED** [review]
Custom hooks are SO much neater than what we had before. I think we can clean up that logic even further, but that's for another ticket. 

This all looks good. Let's get it on staging and check that it's all good. Maybe send it around to people as well. :+1:

**Pull request merged**